util/src/main/java/io/kubernetes/client/ProtoClient.java

```diff
@@ -69,43 +103,98 @@ public void setApiClient(ApiClient apiClient) {
      * @param path The URL path to call (e.g. /api/v1/namespaces/default/pods/pod-name)
      * @return A Message of type T
      */
-    public <T extends Message> T list(T.Builder listObj, String path) throws ApiException, IOException {
+    public <T extends Message> ObjectOrStatus<T> list(T.Builder listObj, String path) throws ApiException, IOException {
         return get(listObj, path);
     }
 
     /**
+     * Create a Kubernetes API object using protocol buffer encoding. Performs a POST
+     * @param obj The object to create
+     * @param path The URL path to call
+     * @param apiVersion The api version to use
+     * @param kind The kind of the object
+     * @return The response received.
+     */
+    public <T extends Message> ObjectOrStatus<T> create(T obj, String path, String apiVersion, String kind)
+            throws ApiException, IOException {
+        return request(obj.newBuilderForType(), path, "POST", obj, apiVersion, kind);
+    }
+
+    /**
+     * Update a Kubernetes API object using protocol buffer encoding. Performs a PUT
+     * @param obj The object to create
+     * @param path The URL path to call
+     * @param apiVersion The api version to use
+     * @param kind The kind of the object
+     * @return The response received.
+     */
+    public <T extends Message> ObjectOrStatus<T> update(T obj, String path, String apiVersion, String kind)
+            throws ApiException, IOException {
+        return request(obj.newBuilderForType(), path, "PUT", obj, apiVersion, kind);
+    }
+
+    /**
+     * Delete a kubernetes API object using protocol buffer encoding.
+     * @param builder The builder for the response
+     * @param path The path to call in the API server
+     * @return The response received
+     */
+    public <T extends Message> ObjectOrStatus<T> delete(T.Builder builder, String path) throws ApiException, IOException {
+        return request(builder, path, "DELETE", null, null, null);
+    }
+
+    /**
      * Generic protocol buffer based HTTP request.
      * Not intended for general consumption, but public for advance use cases.
      * @param builder The appropriate Builder for the object receveived from the request.
      * @param method The HTTP method (e.g. GET) for this request.
      * @param path The URL path to call (e.g. /api/v1/namespaces/default/pods/pod-name)
      * @return A Message of type T
```

Comment:
add documentation for new fields body, apiVersion, kind?

Reply:
done.